### PR TITLE
[smart_holder] clang-tidy fixes (related to recent clang-tidy changes on master).

### PR DESCRIPTION
**Pull request by rwgk**
Also adding `--keep-going` to `cmake` `build` command, to get all clang-tidy errors at once.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

File: .github/workflows/format.yml [hunk 1]
```diff
@@ -44,4 +44,4 @@ jobs:
         -DCMAKE_CXX_STANDARD=17
 
     - name: Build
-      run: cmake --build build -j 2
+      run: cmake --build build -j 2 -- --keep-going
```
Comment (Skylion007) [inline .github/workflows/format.yml:47]: Thanks. It was indeed super annoying to not get all the clang-tidy errors.

**Comment by rwgk** [comment]
Close-Reopen to retrigger GitHub Actions. There were many "cannot find linux asset" or "win32 asset" errors.

**Comment by rwgk** [comment]
Ignoring two notorious flakes: 1. macos runner not available, 2. #2995.

File: .github/workflows/format.yml (hunk 1 above)
Comment (rwgk) [reply .github/workflows/format.yml:47]: @Skylion007 Do you prefer to adopt this in your PR #3051, or do you think it's better if I open a mini PR to add this to master?

Comment (Skylion007) [reply .github/workflows/format.yml:47]: Adopting

**Pull request merged**